src/librbd/ExclusiveLock.cc

```diff
@@ -40,9 +40,12 @@ bool ExclusiveLock<I>::accept_requests(int *ret_val) const {
   Mutex::Locker locker(ML<I>::m_lock);
 
   bool accept_requests = (!ML<I>::is_state_shutdown() &&
+                          !ML<I>::is_state_pre_releasing() &&
                           ML<I>::is_state_locked() &&
```

Comment:
@dillaman I can't understand this. `is_state_locked()` ensures the only allowed state is `SATE_LOCKED`, so additionally testing that it is not `STATE_PRE_RELEASING` does not make any difference?